src/scripts/getRefreshToken.ts

```diff
@@ -1,43 +1,45 @@
 import fetch from "node-fetch";
-import { DesktopIAPOptions, Fetcher } from "../types";
+import { DesktopIAPOptions } from "../types";
+import url from "url";
 
 export function openBrowser(browser: (url: string) => void, url: string): void {
   browser(url);
 }
 
 export async function getRefreshToken(
-  browser: (url: string) => void,
-  inputHandler: (question: string) => Promise<string>,
-  fetcher: Fetcher = fetch,
+  redirectUrl: string,
+  fetcher: Function = fetch,
   options: DesktopIAPOptions,
+  redirectUri: string,
 ): Promise<string> {
-  const url = `https://accounts.google.com/o/oauth2/v2/auth?client_id=${options.clientId}&response_type=code&scope=openid%20email&access_type=offline&redirect_uri=urn:ietf:wg:oauth:2.0:oob`;
-
-  console.log("Open this URL if your browser didn't open automatically\n", url);
-
-  openBrowser(browser, url);
-
-  // await input from user
-  const loginToken = await inputHandler(
-    "Please input the token you received in the browser",
-  );
-
-  const oauthTokenBaseUrl = "https://www.googleapis.com/oauth2/v4/token";
-
-  const body = {
-    code: loginToken,
-    client_id: options.clientId,
-    client_secret: options.clientSecret,
-    redirect_uri: "urn:ietf:wg:oauth:2.0:oob",
-    grant_type: "authorization_code",
-  };
-
-  const request = await fetcher(oauthTokenBaseUrl, {
-    body: JSON.stringify(body),
-    method: "POST",
-  });
-
-  const response = await request.json();
-
-  return String(response["refresh_token"]);
+  let queryParam = url.parse(redirectUrl, true).query;
+  const loginToken = queryParam.code;
+  if (queryParam.error) {
```

Comment:
Here as well, please.

Reply:
There are some known error codes, but Google could always come with new one. I could check ASCII characters. If error contains not allowed characters I will not write it into console? That do not make much sense....